mne/preprocessing/maxwell.py

```diff
@@ -102,13 +114,51 @@ def maxwell_filter(raw, origin=(0, 0, 40), int_order=8, ext_order=3,
     pS_tot = pinv(S_tot, cond=1e-15)
     # Compute multipolar moments of (magnetometer scaled) data (Eq. 37)
     mm = np.dot(pS_tot, data * coil_scale[:, np.newaxis])
-    # Reconstruct data from internal space (Eq. 38)
-    recon = np.dot(S_in, mm[:S_in.shape[1], :])
 
-    # Return reconstructed raw file object
-    raw_sss = _update_sss_info(raw.copy(), origin, int_order, ext_order,
+    # Reconstruct data from internal space (Eq. 38)
+    recon = np.dot(S_in, mm[:S_in.shape[1], :]) / coil_scale[:, np.newaxis]
+
+    raw_sss = raw.copy()
+    if st_dur is not None:
+        if st_dur > times[-1]:
+            raise ValueError('st_dur (%0.1fs) longer than length of signal in '
+                             'raw (%0.1fs).' % (st_dur, times[-1]))
+        logger.info('Processing data using tSSS with st_dur=%s' % st_dur)
+        # Reconstruct data from external space and compute residual
+        recon_out = np.dot(S_out, mm[S_in.shape[1]:, :]) /\
+            coil_scale[:, np.newaxis]
+        resid = data - (recon + recon_out)
+
+        # Generate time points to break up data in to windows
+        lims = raw.time_as_index(np.arange(times[0], times[-1], st_dur))
+        len_last_buf = raw.times[-1] - raw.index_as_time(lims[-1])[0]
+        if len_last_buf >= st_dur:
+            lims = np.concatenate([lims, [len(raw.times)]])
+        else:
+            # Truncate signal if final window is not present
+            raw_sss.crop(lims[0] / raw.info['sfreq'],
+                         lims[-1] / raw.info['sfreq'], copy=False)
```

Comment:
I talked it over with @staulu, and he thinks we should probably avoid truncation by default like you suggested. Currently, Elekta will just process the last window as long as it's over 500 samples. Samu thinks we should probably specify that based on a length of time (rather than number of samples), so maybe we can come up with a reasonable number and throw a warning if the last window is shorter than that threshold, but go ahead and process it anyways.

Samu also mentioned that it might be helpful for a suggested window length in some cases. As in, we might output to the log that a 16s window would work better for a 32s piece of data when the user requested an 18s window. Would that be worth adding? I'm not sure I can come up with a clean way to calculate that though for any given situation...